Makefile

```diff
@@ -35,6 +35,7 @@ clean:
 	(cd ${DIR}; cd Source; rm -rf Dafny/bin Dafny/obj DafnyDriver/bin DafnyDriver/obj DafnyRuntime/obj DafnyRuntime/bin DafnyServer/bin DafnyServer/obj DafnyPipeline/obj DafnyPipeline/bin DafnyCore/obj DafnyCore/bin)
 	(cd ${DIR} ; dotnet build Source/Dafny.sln -v:q --nologo -target:clean )
 	make -C ${DIR}/Source/DafnyCore -f Makefile clean
+	(cd ${DIR}/Source/Dafny && rm -rf Scanner.cs Parser.cs obj )
```

Comment:
Why do you need the -r and the -f flag? -r is for recursive, and -f is for force, and I don't think you need either of these.

Reply:
The -r is because obj is a directory; the -f is because the arguments might not exist.
The reason for this addition is that the location of the Scanner.cs and Parser.cs files changed at some point. If one switches branches, does a `make clean exe`, errors or an incorrect build will result because of the obsolete Scanner or Parser. I've been caught be this many times and I often have forgotten the problem